Feature: Landing Page

renamrsh opened the pull request:
This PR adds the **initial landing page design**. 
The layout is responsive, however all links currently redirect to the top, instead of their intended sections/pages.

- Any **feedback** on the code is welcome :)
- Let me know if any refinements are needed.

renamrsh committed:
Creating page layout, basic view of header and navigation sections.

renamrsh committed:
Hero seciton + responsive: hero & navigation & header

renamrsh committed:
Features section + responsive

renamrsh committed:
Products section + responsive

renamrsh committed:
Advantages section + responsive

renamrsh committed:
Testimonials and Blog sections, footer + responsive
& fix (table column spaces, sections width, advantages childs wrap)

renamrsh committed:
Small fixes: testimonials slider scroll, responsive bugs (testimonial, footer, table, header, nav) + Refactored CSS for better readability

renamrsh commented:
Hi @pkarpovich! Could you please take a look at this PR when you get a chance? I would be grateful for any feedback or suggestions for improvements. Thanks.

renamrsh committed:
Small fixes html/css 2

Remove majority of style attributes from html, change grid into flex for features section, change tittle.

pkarpovich commented:
## Overall Impression

The site looks visually appealing and has a good structure, but there are several issues that should be addressed to improve code quality and CSS organization.

## Issues

### 1. Excessive Use of ID Selectors

Problem: Overuse of ID selectors significantly reduces the maintainability and scalability of CSS

```css
/* Before */
#heroLeft { width: 50%; }

/* After */
.hero__left { width: 50%; }
```

Problems with this approach:
- Creates unnecessarily high specificity, as IDs have higher specificity than classes
- Makes styles difficult to override downstream
- Reduces component reusability
- Violates CSS best practices for scalable stylesheets

Recommendation:
- Refactor to use primarily class-based selectors
- Reserve IDs for JavaScript hooks
- Consider adopting a methodology like BEM, SMACSS, or CUBE CSS

### 2. JavaScript Issues

```javascript
// Problem: DOM manipulation without error handling
const carousel = document.getElementById("slider");
const arrowBtns = document.querySelectorAll("#wrapper .arrow");
const firstCardWidth = carousel.querySelector("figure").offsetWidth;
```

No error handling if elements aren't found, which could break the carousel functionality. Also, the script is embedded directly in the HTML rather than separated into an external file.

### 3. HTML Structure

```html
<!-- Problem: Typo in navigation -->
<a href="#">Contatct</a>
<!-- Fix to: -->
<a href="#">Contact</a>

<!-- Problem: Inline styles -->
<p style="font-size: 24px;">+</p>
<!-- Move to CSS class -->
```

Multiple instances of inline styles should be moved to the stylesheet

### 4. CSS Organization

```css
/* Problem: Inconsistent naming conventions */
#nContainer { /* camelCase with ID selector */ }
.btnNotFillBlue { /* camelCase */ }
.arrowUp { /* camelCase */ }
#hStocks div { /* element selector with ID */ }

/* Problem: CSS variables defined but not fully utilized */
:root {
  --white: #ffffff;
  --blue: #2F80ED;
  /* These colors are defined but many hardcoded colors still exist in the CSS */
}
```

The CSS mixes different naming conventions (camelCase, kebab-case) and selector types. Many color values are hardcoded rather than using the defined CSS variables.

### 5. Performance Concerns

The site loads multiple SVG files separately, which increases HTTP requests:

```html
<img src="public/heroPhones.svg" alt="Application screenshots">
<img src="public/feature1IconPC.svg" alt="Pc icon">
<img src="public/feature2IconSettings.svg" alt="Settings icon">
<!-- Multiple separate SVG files cause performance issues -->
```

Consider creating an SVG sprite or using an icon font. Additionally, images are missing width/height attributes, which could cause layout shifts.

### 6. Optimize images:
   Add width/height attributes and lazy loading.

Overall, the codebase is on the right track but needs small refinements

pkarpovich reviewed (changes requested):
Useful links that could help enhance points from the previous comment

https://dev.to/clairecodes/reasons-not-to-use-ids-in-css-4ni4
https://medium.com/design-bootcamp/introduction-to-bem-block-element-modifier-a-beginners-guide-35e267b6c39e
https://www.builder.io/blog/fast-images
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_cascade/Specificity

pkarpovich commented:
Regarding your issue with the testimonial slider.

**Root cause:**
Looking at your slider code, I can see that you're using the raw width of a figure element (`firstCardWidth`) for the scroll amount:

```javascript
const firstCardWidth = carousel.querySelector("figure").offsetWidth;
// ...
carousel.scrollLeft += firstCardWidth;
```

The problem is that this doesn't account for the full layout context:
1. Your CSS uses a grid layout with percentage-based widths (`grid-auto-columns: 25%`)
2. There are gaps between cards (`gap: 2px`)
3. You have padding around the carousel (`padding: 5px`)

**Suggested approach:**
Rather than using the raw figure width, calculate the proper scroll amount by considering the total width of the carousel and the number of items. This way, the slider will advance exactly one full card, regardless of responsive breakpoints.

The solution would involve:
1. Calculate the total scrollable width
2. Divide by the number of items to get the exact width per card (including gaps)
3. Use this value for scrolling

This approach will automatically adjust to your responsive breakpoints where the number of visible cards changes.

Let me know if you need any clarification on this approach!

renamrsh committed:
Solve code quality & performance issues + fix bugs

Html file: fix typo, moving inline styles to css, adding width & height & lazy loading attributes to img
Js file: creating file and moving slider inside, handling elements, fixing testimonial slider problem
Public folder: adding stars image into sprite.svg

renamrsh committed:
Replace CSS with SCSS using BEM

Changed id selectors to classes, leaving only required for js hooks. Adopted BEM naming methodology. Reduced amount of media queries.

renamrsh committed:
Change files location
